api/pkg/s3/api-response.go

```diff
@@ -0,0 +1,94 @@
+/*
+ * Minio Cloud Storage, (C) 2015, 2016 Minio, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package s3
+
+import (
+	"encoding/xml"
+	"net/http"
+
+	"github.com/emicklei/go-restful"
+	. "github.com/journeymidnight/yig/error"
+	"github.com/journeymidnight/yig/helper"
+)
+
+const (
+	timeFormatAMZ = "2006-01-02T15:04:05.000Z" // Reply date format
+)
+
+// writeErrorResponse write error headers
+// w http.ResponseWriter, r *http.Request
+func WriteErrorResponse(request *restful.Request, response *restful.Response, err error) {
+	WriteErrorResponseHeaders(response, err)
+	WriteErrorResponseNoHeader(response, request, err, request.Request.URL.Path)
+}
+
+func WriteErrorResponseHeaders(response *restful.Response, err error) {
+	var status int
+	apiErrorCode, ok := err.(ApiError)
+	if ok {
+		status = apiErrorCode.HttpStatusCode()
+	} else {
+		status = http.StatusInternalServerError
+	}
+	helper.Logger.Println(20, "Response status code:", status, "err:", err)
+
+	//ResponseRecorder
+	//w.(*ResponseRecorder).status = status
+
+	response.WriteHeader(status)
+}
+
+func WriteErrorResponseNoHeader(response *restful.Response, request *restful.Request, err error, resource string) {
+	// HEAD should have no body, do not attempt to write to it
+	if request.Request.Method == "HEAD" {
+		return
+	}
+
+	// Generate error response.
+	errorResponse := ApiErrorResponse{}
+	apiErrorCode, ok := err.(ApiError)
+	if ok {
+		errorResponse.AwsErrorCode = apiErrorCode.AwsErrorCode()
```

Comment:
Why is there AwsErrorCode() ??

Reply:
Thanks for your review,  The error code  defined by aws from link "https://docs.aws.amazon.com/AmazonS3/latest/API/ErrorResponses.html"  is used to return client